src/forms/form-fields/helpers/IconPickerDialog/IconPickerCustomTab.js

```diff
@@ -0,0 +1,210 @@
+import React, { Component } from 'react';
+import Button from 'd2-ui/lib/button/Button';
+import RaisedButton from 'material-ui/RaisedButton';
+import PropTypes from 'prop-types';
+import Divider from 'material-ui/Divider';
+import TextField from 'material-ui/TextField/TextField';
+import { fileToBase64 } from './fileToBase64.js';
+import { uploadIcon } from './uploadIcon.js';
+import ErrorMessage from 'd2-ui/lib/messages/ErrorMessage.component';
+import classes from 'classnames';
+
+const styles = {
+    selectFileButton: {
+        backgroundColor: '#ff9800',
+        color: '#fff',
+        textAlign: 'center',
+        position: 'relative',
+        minWidth: 129,
+        height: 36,
+        lineHeight: 2.5,
+        boxShadow: '0 1px 6px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.12)',
+        cursor: 'pointer',
+        textTransform: 'uppercase',
+    },
+    iconFileInput: {
+        display: 'none',
+    },
+};
+
+export class IconPickerCustomTab extends Component {
+    constructor(props, context) {
+        super(props, context);
+
+        this.t = context.d2.i18n.getTranslation.bind(context.d2.i18n);
+        this.d2 = context.d2;
+        this.state = {
+            iconFile: null,
+            iconFileBase64: null,
+            iconMetadata: {
+                key: '',
+                description: '',
+                keywords: [],
+            },
+            uploading: false,
+            uploadError: null,
+        };
+    }
+
+    handleIconMetadataChange = event => {
+        const value = event.target.value && event.target.value.trim();
+        const name = event.target.name;
+
+        this.setState(prevState => ({
+            iconMetadata: {
+                ...prevState.iconMetadata,
+                [name]: value,
+            },
+        }));
+    };
+
+    handleIconKeywordsChange = event => {
+        const keywords = event.target.value
+            .split(',')
```

Comment:
maybe we can be lenient here and split by new lines too - this seems like my default go-to (and the hint text is not very obvious)
```suggestion
          .split(/[,(\r?\n)]+/)
```

Reply:
Added returns. Not sure about the `()` and `?`, though?

Reply:
the `?` as `\r` is present on windows, but not on linux/mac .. but I think your code would work anyhow since there is a trim after that will get rid of either \r or \n (or both) .. so you could split by just `\n`

Reply:
As far as I understand the `?` and `()` would be replaced literally. Eg. split on `?` and parenthesis?

Reply:
```js
`is this
a string?
Yes?Iam,am(split)`.split(/[,(\r?\n)]+/)
```

Results in:
```js
['is this', 'a string', 'Yes', 'Iam', 'am', 'split', '']
```